commodore/inventory/lint.py

```diff
@@ -92,14 +103,56 @@ def _lint_directory(cfg: Config, path: Path, lintfunc: LintFunc) -> int:
             if cfg.debug:
                 click.echo(f"> Skipping hidden directory entry {dentry}")
             continue
+        if dentry.absolute() in ignore_paths:
+            if cfg.debug:
+                click.echo(f"> Skipping ignored directory entry {dentry}")
+            continue
         if dentry.is_dir():
-            errcount += _lint_directory(cfg, dentry, lintfunc)
+            errcount += _lint_directory(cfg, dentry, ignore_paths, lintfunc)
         else:
             errcount += _lint_file(cfg, dentry, lintfunc)
     return errcount
 
 
-def run_linter(cfg: Config, path: Path, lintfunc: LintFunc) -> int:
+def _read_ignore_patterns_from_file(path: Path) -> set[str]:
+    ignore_patterns = set()
+    if path.is_file():
+        with open(path, "r", encoding="utf-8") as ifile:
+            for p in ifile.readlines():
+                ignore_patterns.add(p.strip())
+
+    return ignore_patterns
+
+
+def _render_ignore_patterns(
+    base_dir: Path, ignore_patterns: tuple[str, ...]
+) -> set[Path]:
+    """Use `glob.glob()` to render the paths to ignore from `ignore_patterns`.
+    All patterns are rooted at `base_dir`.
+
+    If pattern doesn't start with /, we will treat it as matching any prefix in
+    `base_dir`."""
+
+    # Read additional ignore patterns from `<base_dir>/.commodoreignore`, if the file
+    # exists
+    _ignore_patterns = set(p for p in ignore_patterns)
```

Comment:
Maybe this is some python thing I don't understand but why the list comprehension?

Reply:
This was a leftover from some previous version of the implementation. It's not required at all.